sample-app/src/components/InputDropdown.tsx

```diff
@@ -78,18 +79,20 @@ export default function InputDropdown({
   const [childrenKey, setChildrenKey] = useState(0);
   const [screenReaderKey, setScreenReaderKey] = useState(0);
 
-  const inputRef = useRef<HTMLInputElement>(document.createElement('input'));
-  const dropdownRef = useRef<HTMLDivElement>(document.createElement('div'));
+  const inputRef = useRef<HTMLInputElement>(null);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const inputDropdownRef = useRef<HTMLDivElement>(null);
 
   if (!shouldDisplayDropdown && screenReaderKey) {
     setScreenReaderKey(0);
   }
 
-  const childrenArray = React.Children.toArray(children);
-  const childrenWithProps = childrenArray.map((child, index) => {
+  let numSections = 0;
+  const childrenWithProps = recursivelyMapChildren(children, (child, index) => {
```

Comment:
For my own edification, why do we need this recursion here? Does this mean that if we were to render 10 entity previews, say, each would be registered as a "child with props"? Does `childrenWithProps` need to represent each individual Component rendered inside the dropdown as opposed to the actual top-level sections?

Reply:
this is needed so that InputDropdown can handle DropdownSection children that are not direct children, i.e. if you wanted to drop it in a div or fragment
something like
```tsx
<InputDropdown>
  <>
    <DropdownSection/>
  </>
</InputDropdown>
```
with the above example InputDropdown would not see any DropdownSections without recursing through children.
I'm unaware whether the intention was for DropdownSection to only be specified as top level children, but that feels a little restrictive. 
For your 2nd/3rd question, every child of InputDropdown is included in `childrenWithProps`, not just children of type DropdownSection. For instance, in the above example `childrenWithProps` only contains the fragment.